static/gui/static/js/components/blockly.js

```diff
@@ -88,39 +80,40 @@ class PythonTextBox extends React.Component {
     this.setState({ filename: event.target.value });
   }
 
-  /* Target function for detected text changes in the editing box.
-     Update this.state with the current text. */
-  handleScriptChange(event) {
-    this.setState({ pythonTextBoxCode: event.target.value });
-    if (this.state.coding_start == -1) {
-      this.setState({ coding_start: new Date().getTime()})
-    }
-  }
-
-  handleTab(event) {
-    if (event.keyCode==9){
-      event.preventDefault();
-      document.getElementById("textarea").value = this.state.pythonTextBoxCode+'    ';
-      this.setState({ pythonTextBoxCode: event.target.value });
-    }
-  }
-
+  /* Target function for the button "Choose File". Uploads the file specified
+        and updates the editor-code. */
   upload(event) {
     var _this = this;
     var file = event.target.files[0];
     var reader = new FileReader();
     reader.onload = function (event) {
-      _this.state.pythonTextBoxCode = event.target.result;
-      document.getElementById("textarea").value = event.target.result;
+      _this.getEditor().setValue(event.target.result);
     };
-    this.setState({ pythonTextBoxCode: generatedPythonFromBlocklyBox.innerText });
     reader.readAsText(file);
   }
 
   view_history(event) {
     window.open("http://127.0.0.1:5000/program/")
   }
 
+  /* Target function for the button "Download". Downlaods the code in the
+      editor as a python file with name as specified */
+  download_python(event) {
+    console.log("download listener");
+    event.preventDefault();
```

Comment:
Could you add a comment explaining why you call "event.preventDefault()"?

Reply:
Wait so why is the event.preventDefault() called?  Could you add a comment explaining this?